tools/generate_hotkeys.py

```diff
@@ -1,39 +1,141 @@
 #!/usr/bin/env python3
+import argparse
+import re
+from collections import defaultdict
 from pathlib import Path, PurePath
+from typing import Dict, List, Tuple
 
 from zulipterminal.config.keys import HELP_CATEGORIES, KEY_BINDINGS
 
 
-categories = {
-    category: {
-        item["help_text"]: item["keys"]
-        for item in KEY_BINDINGS.values()
-        if item["key_category"] == category
-    }
-    for category in HELP_CATEGORIES.keys()
-}
 OUTPUT_FILE = Path(__file__).resolve().parent.parent / "docs" / "hotkeys.md"
 SCRIPT_NAME = PurePath(__file__).name
 
-with open(OUTPUT_FILE, "w") as mdFile:
-    mdFile.write(
+# Exclude keys from duplicate keys checking
+KEYS_TO_EXCLUDE = ["q", "e", "m", "r"]
+
+
+def main(check_only: bool = False) -> None:
+    if check_only:
+        lint_hotkeys_file()
+    else:
+        generate_hotkeys_file()
+
+
+def lint_hotkeys_file() -> None:
+    """
+    First lint config/keys.py file for key description and key duplication, then compare if in sync with existing hotkeys.md file.
+    """
+    hotkeys_file_string = get_hotkeys_file_string()
+    # To lint keys description and duplicate keys
+    error_flag = 0
+    help_text_style = re.compile(r"^[^\d.]*$")
+    categories = read_help_categories()
+    for action in HELP_CATEGORIES.keys():
+        check_duplicate_keys_list: List[str] = []
+        for help_text, key_combinations_list in categories[action]:
+            check_duplicate_keys_list.extend(key_combinations_list)
+            various_key_combinations = " / ".join(key_combinations_list)
+            # Check description style
+            if not re.match(help_text_style, help_text):
+                print(
+                    f"Description - ({help_text}) for key combination - [{various_key_combinations}] should contain only alphabets, spaces and special characters except ."
+                )
+                error_flag = 1
+        # Check key combination duplication
+        check_duplicate_keys_list = [
+            key for key in check_duplicate_keys_list if key not in KEYS_TO_EXCLUDE
+        ]
+        duplicate_keys = [
+            key
+            for key in check_duplicate_keys_list
+            if check_duplicate_keys_list.count(key) > 1
+        ]
+        if len(duplicate_keys) != 0:
+            print(
+                f"Duplicate key combination for keys {duplicate_keys} for category ({HELP_CATEGORIES[action]}) detected"
+            )
+            error_flag = 1
+    if error_flag == 1:
+        print("Rerun this command after resolving errors in config/keys.py")
+    else:
+        print("No hotkeys linting errors")
+        if not compare_hotkeys_file(hotkeys_file_string):
```

Comment:
I have used return value for ``compare_hotkeys_file`` to output different statement based on return value.